framework/src/meshgenerators/CutMeshByPlaneGenerator.C

```diff
@@ -0,0 +1,502 @@
+      std::vector<std::vector<unsigned int>> rotated_tet_face_indices;
+      std::vector<std::vector<const Node *>> optimized_node_list;
+      MooseMeshElementConversionUtils::prismNodesToTetNodesDeterminer(
+          new_elems_nodes, rotated_tet_face_indices, optimized_node_list);
+
+      for (unsigned int i = 0; i < optimized_node_list.size(); i++)
+      {
+        auto new_elem_tet4 = std::make_unique<Tet4>();
+        new_elem_tet4->set_node(0) = const_cast<Node *>(optimized_node_list[i][0]);
+        new_elem_tet4->set_node(1) = const_cast<Node *>(optimized_node_list[i][1]);
+        new_elem_tet4->set_node(2) = const_cast<Node *>(optimized_node_list[i][2]);
+        new_elem_tet4->set_node(3) = const_cast<Node *>(optimized_node_list[i][3]);
+        new_elem_tet4->subdomain_id() = mesh.elem_ptr(elem_id)->subdomain_id();
+        elems_tet4.push_back(mesh.add_elem(std::move(new_elem_tet4)));
+      }
+    }
+    else if (tet4_nodes_inside_plane.size() == 3 && tet4_nodes_outside_plane.size() == 1)
+    {
+      std::vector<const Node *> new_plane_nodes;
+      // 3 smaller Tet4 elements are created
+      for (const auto & tet4_node_inside_plane : tet4_nodes_inside_plane)
+      {
+        new_plane_nodes.push_back(nonDuplicateNodeCreator(
+            mesh,
+            new_on_plane_nodes,
+            pointPairPlaneInterception(
+                *tet4_node_inside_plane, *tet4_nodes_outside_plane[0], plane_point, plane_normal)));
+      }
+      std::vector<const Node *> new_elems_nodes = {tet4_nodes_inside_plane[0],
+                                                   tet4_nodes_inside_plane[1],
+                                                   tet4_nodes_inside_plane[2],
+                                                   new_plane_nodes[0],
+                                                   new_plane_nodes[1],
+                                                   new_plane_nodes[2]};
+      std::vector<std::vector<unsigned int>> rotated_tet_face_indices;
+      std::vector<std::vector<const Node *>> optimized_node_list;
+      MooseMeshElementConversionUtils::prismNodesToTetNodesDeterminer(
+          new_elems_nodes, rotated_tet_face_indices, optimized_node_list);
+
+      for (unsigned int i = 0; i < optimized_node_list.size(); i++)
+      {
+        auto new_elem_tet4 = std::make_unique<Tet4>();
+        new_elem_tet4->set_node(0) = const_cast<Node *>(optimized_node_list[i][0]);
+        new_elem_tet4->set_node(1) = const_cast<Node *>(optimized_node_list[i][1]);
+        new_elem_tet4->set_node(2) = const_cast<Node *>(optimized_node_list[i][2]);
+        new_elem_tet4->set_node(3) = const_cast<Node *>(optimized_node_list[i][3]);
+        new_elem_tet4->subdomain_id() = mesh.elem_ptr(elem_id)->subdomain_id();
+        elems_tet4.push_back(mesh.add_elem(std::move(new_elem_tet4)));
+      }
+    }
+    else if (tet4_nodes_inside_plane.size() == 1 && tet4_nodes_outside_plane.size() == 1)
```

Comment:
this case is not hit by any tests based on the coverage report
we should cover it